Remove trace from link script

WalterKolczynski-NOAA opened the pull request:
**Description**
Removes the trace from the link script. Since this makes the script silent (including some errors), some messages are added to give feedback about the scripts success or failure.

UFS Utils' [link_fixdirs.sh](https://github.com/ufs-community/UFS_UTILS/blob/develop/fix/link_fixdirs.sh) script turns the trace on when it runs, so its STDERR is thrown away.

Also corrects a bug where the case of `$RUN_ENVIR` set did not match the case it was compared against.

Fixes #1044

**Type of change**

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**
- [x] `link_workflow.sh -h` on Orion
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

WalterKolczynski-NOAA committed:
Remove trace from link script

Fixes #1044

RussTreadon-NOAA reviewed (approved):
Thank you, @WalterKolczynski-NOAA , for making this change.

WalterKolczynski-NOAA commented:
There are other issues with turning off the trace. This is going to need more thought.

WalterKolczynski-NOAA committed:
Add messages when link script exits, fix case error

Since removing the trace rendered the link script completely silent,
added some messages when it ends. A message reporting the successful
completion is added to the bottom. For errors, an ERR trap is added
that reports the error, along with the error code and line number.

Since UFS utils' `link_fixdirs.sh` turns the trace on, throw away its
STDERR. This isn't an ideal solution, but will do for now without
changes to UFS utils.

Also changed the case of `$RUN_ENVIR` to lowercase to match the
strings that are actually checked.

Refs: #1044

RussTreadon-NOAA reviewed (approved):
Approve assuming that updated script has been tested and found to work as intended.
